pkg/apis/extensions/v1alpha1/type_network.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Network is the specification for cluster networking.
+type Network struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NetworkSpec   `json:"spec"`
+	Status NetworkStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NetworkList is a list of Network resources.
+type NetworkList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// Items is the list of Networks.
+	Items []Network `json:"items"`
+}
+
+// NetworkSpec is the spec for an Network resource.
+type NetworkSpec struct {
+	// DefaultSpec is a structure containing common fields used by all extension resources.
+	DefaultSpec `json:",inline"`
+
+	// ClusterCIDR defines the CIDR that will be used for pods.
+	ClusterCIDR string `json:"clusterCIDR"`
+	// ServiceCIDR defines the CIDR that will be used for services.
+	ServiceCIDR string `json:"serviceNetworkCIDR"`
+	// NetworkConfig contains plugin-specific configuration.
+	// +optional
+	NetworkConfig *runtime.RawExtension `json:"networkConfig,omitempty"`
+}
+
+// NetworkStatus is the status for an Network resource.
+type NetworkStatus struct {
+	// DefaultStatus is a structure containing common fields used by all extension resources.
+	DefaultStatus `json:",inline"`
+
+	// NetworkStatus contains plugin-specific output.
+	// +optional
+	NetworkStatus *runtime.RawExtension `json:"providerStatus,omitempty"`
```

Comment:
```suggestion
	ProviderStatus *runtime.RawExtension `json:"providerStatus,omitempty"`
```

Reply:
what does provider stand for here, it made sense in the cloud-provider context but do you think it still makes sense here?